torch/distributions/half_normal.py

```diff
@@ -0,0 +1,39 @@
+import math
+from torch.distributions import constraints
+from torch.distributions.transforms import AbsTransform
+from torch.distributions.normal import Normal
+from torch.distributions.transformed_distribution import TransformedDistribution
+
+
+class HalfNormal(TransformedDistribution):

```

Comment:
I agree this one might not be very useful. @jpchen is working on a `HalfCauchy` which is actually commonly used as a prior on variances especially in linguistics and is worth maintaining. We could use `HalfCauchy` as the canonical demonstration of `AbsTransform` and add `TruncatedNormal` in a later PR (`TruncatedNormal` is actually really tricky to get right, due to reparameterized gradients). @alicanb WDYT?

Reply:
Works for me 👍 I've been thinking about how to handle `TruncatedNomal` but we can discuss that later. In the meantime, we can remove `HalfNormal` and ask @jpchen to push an implementation of `HalfCauchy` with `Transforms` to this branch (Nice thing about `HalfCauchy` is it uses both `AffineTransform` and AbsTransform`, so it will be a perfect example)

Reply:
i can do that in a separate PR, i have an outstanding one in the probtorch fork which i can rewrite after this is merged, since i would need to look into the semantics of the transformations